### PR TITLE
Migration system test

### LINKED ISSUE #3310: Esti system test for DB migration to KV Store
* Esti system test to verify a system wide migration to KV, for correctness
  * Can also be used to cover the migration from the performance aspect by (currently manually) checking the time it takes
  * As the migration process will increase over time, as we transfer more parts to KV, the time to migrate cannot be well defined
  * We will use this to identify "unreasonable" migration time

* The main idea is to have a system running on SQL DB, performing various lakefs actions [TBD] and bringing lakefs to a certain state. Then lakefs is stopped, a migration to KV takes place and lakefs is restarted with KV enabled. The testing flow should continue at this point, seamlessly
* **Note**: Currently, Esti tests rely on lakefs being up and running prior to the test itself. This task needs to add support for stopping/starting lakefs as part of the test
* The time of the migration process should be "benchmarked" as part of this test. Initially, this can be done by manually verifying the migration time
* What is the sufficient dataset prior to migration?

### PR DESCRIPTION
**Pull request by itaiad200**
closes #3310

### CODE REVIEW

File: esti/migrate_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	if postMigrate {
+		postMigrateTests(t)
+	} else {
+		preMigrateTests(t)
+	}
+}
+
+func preMigrateTests(t *testing.T) {
+	// all pre tests execution
+	t.Run("TestPreMigrateMultipart", testPreMigrateMultipart)
```
Comment (N-o-Z) [inline esti/migrate_test.go:30]: Does it make sense to continue the pre migrate tests if one fails?

File: esti/migrate_test.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+	saveStateInLakeFS(t)
+}
+
+func postMigrateTests(t *testing.T) {
```
Comment (N-o-Z) [inline esti/migrate_test.go:35]: Same as the comment before but for the general usecase.
If preMigrateTests failed - there's no point in running postMigrateTests.
We need to somehow indicate the pre tests failed (maybe via the state file?)

File: esti/migrate_test.go [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+const (
+	migrateMultipartFile     = "multipart_file"
+	migrateMultipartFilepath = mainBranch + "/" + migrateMultipartFile
+	migrateStateRepoName     = "migrate"
+	migrateStateObjectPath   = "state.json"
+)
+
+func testPreMigrateMultipart(t *testing.T) {
```
Comment (N-o-Z) [inline esti/migrate_test.go:86]: Just an idea - not sure how much of a usecase it is:
How about uploading only some of the parts in the old DB format, migrating and then uploading the rest?

File: esti/migrate_test.go [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+	completedParts := uploadMultipartParts(t, logger, resp, parts)
+
+	state.Multipart.Repo = repo
+	state.Multipart.Info = *resp
+	state.Multipart.CompletedParts = completedParts
+	state.Multipart.Content = base64.StdEncoding.EncodeToString(partsConcat)
+}
+
+func testPostMigrateMultipart(t *testing.T) {
```
Comment (N-o-Z) [inline esti/migrate_test.go:113]: I would arrange all the test functions together and order them hierarchically and put the variables declaration at the top of the module

**Review by N-o-Z — CHANGES_REQUESTED** [review]
Very cool! 
Some comments

File: esti/migrate_test.go (hunk 1 above)
Comment (itaiad200) [reply esti/migrate_test.go:30]: We'll have to answer that when we'll have more than a single migration test.

File: esti/migrate_test.go (hunk 2 above)
Comment (itaiad200) [reply esti/migrate_test.go:35]: Right - If Pre test fails -> Esti fails -> flow stops and we'll never reach the post test.

File: esti/migrate_test.go (hunk 3 above)
Comment (itaiad200) [reply esti/migrate_test.go:89]: Done.

**Review by N-o-Z — APPROVED** [review]
Cool!

**Pull request merged**